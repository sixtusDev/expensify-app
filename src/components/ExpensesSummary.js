import React from 'react'
import numeral from 'numeral'
import  { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpensesSummary = (props) => {
  const expenseWord = props.expensesCount === 1 ? 'expense' : 'expenses'
  return (
    <div>
      <h2>
        Viewing {props.expensesCount} {expenseWord} totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}
      </h2>
    </div>
  )
}

const mapStateToProps = (state) => {
  const expenses = selectExpenses(state.expenses, state.filters)
  return {
    expensesCount: expenses.length,
    expensesTotal: selectExpensesTotal(expenses)
  }
}

export default connect(mapStateToProps)(ExpensesSummary)