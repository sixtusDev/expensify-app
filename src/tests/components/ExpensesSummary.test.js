import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should render ExpensesSummary correctly with no expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary expensesCount={0} expensesTotal={0}/>
  )
  expect(wrapper).toMatchSnapshot()
})

test('Should render ExpensesSummary correctly with multiple expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary expensesCount={50} expensesTotal={120003044458}/>
  )
  expect(wrapper).toMatchSnapshot()
})