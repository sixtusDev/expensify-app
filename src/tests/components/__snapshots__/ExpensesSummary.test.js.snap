// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    50
     
    expenses
     totalling 
    $1,200,030,444.58
  </h2>
</div>
`;

exports[`Should render ExpensesSummary correctly with no expenses 1`] = `
<div>
  <h2>
    Viewing 
    0
     
    expenses
     totalling 
    $0.00
  </h2>
</div>
`;
